build-info-extractor/src/main/java/org/jfrog/build/extractor/scan/Artifact.java

```diff
@@ -0,0 +1,60 @@
+package org.jfrog.build.extractor.scan;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author yahavi
+ */
+public class Artifact implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private GeneralInfo generalInfo = new GeneralInfo();
+    private Set<Issue> issues = new HashSet<>();
+    private Set<License> licenses = new HashSet<>();
+
+    // Empty constructor for serialization
+    public Artifact() {
+    }
+
+    public Artifact(GeneralInfo generalInfo, Set<Issue> issues, Set<License> licenses) {
+        this.generalInfo = new GeneralInfo(generalInfo);
```

Comment:
Question - wouldn't a simple assignment be sufficient here?  - 
```this.generalInfo = generalInfo;```
Do we really need a clone of GeneralInfo here?
If so, then the constructor in GeneralInfo which accepts another GeneralInfo may be redundant.

The same question goes for *this.issues* and *this.licenses* in the same method - are we actually aggregating issues and licenses (with addAll) or this is a one time assignment?